### PR TITLE
Gatk cnv

### PR DESCRIPTION
**Pull request by jonca79**

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -9,12 +9,17 @@ default_container: "docker://gmsuppsala/somatic:develop"
...
-
-reference:
-  fasta: "reference/WASH7P.fna"
-  design_bedfile: "reference/twist_DNA_solid.chr1.annotated.bed"
-  
 germline_vcf:
   container: "docker://ensemblorg/ensembl-vep:release_99.0"
+
+cnvkit_call:
+  normal_reference: "/projects/wp1/nobackup/ngs/utveckling/Twist_DNA_DATA/PoN/cnvkit_Twist_PoN.cnn"
```
Comment (Smeds) [inline config/config.yaml:16]: should we maybe remove part of the path, the wp1 stuff. It's very specific to our environment and I guess we will not have that storage path as our final path.

Comment (marrip) [reply config/config.yaml:16]: agreed, this should be an example on how to structure the `config.yaml` and not a specific setup.

File: workflow/rules/gatk_cnv_collect_allelic_counts.smk [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    params:
+        extra=config.get("gatk_cnv_collect_allelic_counts", {}).get("extra", ""),
+    log:
+        "cnv/gatk_cnv_collect_allelic_counts/{sample}_{type}.clean.allelicCounts.tsv.log",
+    benchmark:
+        repeat(
+            "cnv/gatk_cnv_collect_allelic_counts/{sample}_{type}.clean.allelicCounts.tsv.benchmark.tsv",
+            config.get("gatk_cnv_collect_allelic_counts", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("gatk_cnv_collect_allelic_counts", config["default_resources"]).get("threads", config["default_resources"]["threads"])
```
Comment (Smeds) [inline workflow/rules/gatk_cnv_collect_allelic_counts.smk:27]:
```suggestion
    threads: config.get("gatk_cnv_collect_allelic_counts", {}).get("threads", config["default_resources"]["threads"])
    resources:
        threads=config.get("gatk_cnv_collect_allelic_counts", {}).get("threads", config["default_resources"]["threads"]),
        time=config.get("gatk_cnv_collect_allelic_counts", {}).get("time", config["default_resources"]["time"]),
```

File: workflow/rules/gatk_cnv_collect_read_counts.smk [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        mergingRule="OVERLAPPING_ONLY",
+        extra=config.get("gatk_cnv_collect_read_counts", {}).get("extra", ""),
+    log:
+        "cnv/gatk_cnv_collect_read_counts/{sample}_{type}.counts.hdf5.log",
+    benchmark:
+        repeat(
+            "cnv/gatk_cnv_collect_read_counts/{sample}_{type}.counts.hdf5.benchmark.tsv",
+            config.get("gatk_cnv_collect_read_counts", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("gatk_cnv_collect_read_counts", config["default_resources"]).get("threads", config["default_resources"]["threads"])
```
Comment (Smeds) [inline workflow/rules/gatk_cnv_collect_read_counts.smk:27]:
```suggestion
    threads: config.get("gatk_cnv_collect_read_counts", {}).get("threads", config["default_resources"]["threads"])
    resources:
        threads=config.get("gatk_cnv_collect_read_counts", {}).get("threads", config["default_resources"]["threads"]),
        time=config.get("gatk_cnv_collect_read_counts", {}).get("time", config["default_resources"]["time"]),
```

File: workflow/rules/gatk_cnv_denoise_read_counts.smk [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    params:
+        extra=config.get("gatk_cnv_denoise_read_counts", {}).get("extra", ""),
+    log:
+        "cnv/gatk_cnv_denoise_read_counts/{sample}_{type}.clean.denoisedCR.tsv.log",
+    benchmark:
+        repeat(
+            "cnv/gatk_cnv_denoise_read_counts/{sample}_{type}.clean.denoisedCR.tsv.benchmark.tsv",
+            config.get("gatk_cnv_denoise_read_counts", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("gatk_cnv_denoise_read_counts", config["default_resources"]).get("threads", config["default_resources"]["threads"])
```
Comment (Smeds) [inline workflow/rules/gatk_cnv_denoise_read_counts.smk:26]:
```suggestion
    threads: config.get("gatk_cnv_denoise_read_counts", {}).get("threads", config["default_resources"]["threads"])
    resources:
        threads=config.get("gatk_cnv_denoise_read_counts", {}).get("threads", config["default_resources"]["threads"]),
        time=config.get("gatk_cnv_denoise_read_counts", {}).get("time", config["default_resources"]["time"]),
```

File: workflow/rules/gatk_cnv_model_segments.smk [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        outprefix="{sample}_{type}.clean",
+        extra=config.get("gatk_cnv_model_segments", {}).get("extra", ""),
+    log:
+        "cnv/gatk_cnv_model_segments/{sample}_{type}.clean.modelFinal.seg.log",
+    benchmark:
+        repeat(
+            "cnv/gatk_cnv_model_segments/{sample}_{type}.clean.modelFinal.seg.benchmark.tsv",
+            config.get("gatk_cnv_model_segments", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("gatk_cnv_model_segments", config["default_resources"]).get("threads", config["default_resources"]["threads"])
```
Comment (Smeds) [inline workflow/rules/gatk_cnv_model_segments.smk:36]:
```suggestion
    threads: config.get("gatk_cnv_model_segments", {}).get("threads", config["default_resources"]["threads"])
    resources:
        threads=config.get("gatk_cnv_model_segments", {}).get("threads", config["default_resources"]["threads"]),
        time=config.get("gatk_cnv_model_segments", {}).get("time", config["default_resources"]["time"]),
```

**Pull request merged**